### PR TITLE
feat #106277: Feature/leakage ALD sectors

### PR DESCRIPTION
**Pull request by olsoybakk**
## Checklist
- [x] I have run linter.
- [ ] I added some tests. (to be defined further) [optional]
- [ ] I have updated [the wiki.](https://github.com/equinor/echo-maps/wiki) [optional]

## Subsea Leakage styles and sectors
- Added sectors for ALD
- Added tooltip for ALD sectors
- Fixed styling for SSLD (correct number in badges)
- [AB#106277](https://dev.azure.com/EquinorASA/5f193d29-b7e1-4e23-aba0-7eac7e847ab5/_workitems/edit/106277)
- [AB#105775](https://dev.azure.com/EquinorASA/5f193d29-b7e1-4e23-aba0-7eac7e847ab5/_workitems/edit/105775)
- [AB#106209](https://dev.azure.com/EquinorASA/5f193d29-b7e1-4e23-aba0-7eac7e847ab5/_workitems/edit/106209)

## Screenshots
image
image

### CODE REVIEW

File: src/layers/external/common/ColorMapping.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        return;
+    }
+    const sortedEntries = Object.entries(countColorMapping).sort(
+        ([, a], [, b]) => (a.order ?? Infinity) - (b.order ?? Infinity),
+    );
+
+    return Object.fromEntries(sortedEntries);
+}
+
+export function createCountColorMapping(
```
Comment (ErlendAHall) [inline src/layers/external/common/ColorMapping.ts:76]: This function could be simplified by just accepting a list of Features, even if the list only contains 1 indice

File: src/layers/external/common/ColorMapping.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    }
+    return sortCountColorMapping(countColorMapping);
+}
+
+export function getSignificantColorsFromColorMapping(
+    countColorMapping?: CountColorMapping,
+): FilterLabelColorMap {
+    let colors: FilterLabelColorMap | undefined;
+    Object.entries(countColorMapping ?? []).forEach(([key, mapping]) => {
+        if (!mapping || mapping.order === undefined) {
```
Comment (ErlendAHall) [inline src/layers/external/common/ColorMapping.ts:101]: this check is error prone as mapping.order could be a null or other falsey values in the future

Comment (ErlendAHall) [inline src/layers/external/common/ColorMapping.ts:101]: judging from the function, this check should determine if mapping.order is a number, and preferably not a null even though the value coercion should handle that as a number 0

**Review by ErlendAHall — APPROVED** [review]

File: src/layers/external/common/ColorMapping.ts (hunk 1 above)
Comment (olsoybakk) [reply src/layers/external/common/ColorMapping.ts:76]: thanks

**Review by ErlendAHall — APPROVED** [review]

**Pull request merged**